File: model.requisiti/test/it/unibo/ConnTcp.java
package it.unibo;


import unibo.actor22comm.interfaces.Interaction2021;
import unibo.actor22comm.tcp.TcpClientSupport;
import unibo.actor22comm.utils.ColorsOut;

public class ConnTcp implements Interaction2021 {
    private Interaction2021 conn;

    public ConnTcp(String hostAddr, int port) throws Exception{
        conn = TcpClientSupport.connect(hostAddr,port,10);
        ColorsOut.out("ConnTcp createConnection DONE:" + conn, ColorsOut.GREEN);
     }

    @Override
    public void forward(String msg) {
        try {
            //ColorsOut.outappl("ConnTcp forward:" + msg   , ColorsOut.GREEN);
            conn.forward(msg );
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public String request(String msg) throws Exception {
        forward(msg);
        return receiveMsg();
    }

    @Override
    public void reply(String s) throws Exception {
        forward(s);
    }

    @Override
    public String receiveMsg() throws Exception {
          return conn.receiveMsg();
    }

    @Override
    public void close() throws Exception {
        conn.close();
    }


    @Override
    public void sendALine(String s) throws Exception {
        conn.sendALine(s);
    }

    @Override
    public void sendALine(String s, boolean b) throws Exception {
        conn.sendALine(s, b);
    }

    @Override
    public String receiveALine() throws Exception {
        return conn.receiveALine();
    }

    @Override
    public void closeConnection() throws Exception {
        conn.closeConnection();
    }
}

File: wasteservice.shared/src-sprint1/main/kotlin/it/unibo/lenziguerra/wasteservice/utils/ConnTcp.kt
package it.unibo.lenziguerra.wasteservice.utils

import unibo.comm22.interfaces.Interaction2021
import unibo.comm22.tcp.TcpClientSupport
import unibo.comm22.utils.ColorsOut

class ConnTcp(hostAddr: String?, port: Int) : Interaction2021 {
    private val conn: Interaction2021

    init {
        conn = TcpClientSupport.connect(hostAddr, port, 10)
        ColorsOut.out("ConnTcp createConnection DONE:$conn", ColorsOut.GREEN)
    }

    override fun forward(msg: String) {
        try {
            //ColorsOut.outappl("ConnTcp forward:" + msg   , ColorsOut.GREEN);
            conn.forward(msg)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    @Throws(Exception::class)
    override fun request(msg: String): String {
        forward(msg)
        return receiveMsg()
    }

    @Throws(Exception::class)
    override fun reply(s: String) {
        forward(s)
    }

    @Throws(Exception::class)
    override fun receiveMsg(): String {
        return conn.receiveMsg()
    }

    @Throws(Exception::class)
    override fun close() {
        conn.close()
    }

    @Throws(Exception::class)
    override fun sendALine(s: String) {
        conn.sendALine(s)
    }

    @Throws(Exception::class)
    override fun sendALine(s: String, b: Boolean) {
        conn.sendALine(s, b)
    }

    @Throws(Exception::class)
    override fun receiveALine(): String {
        return conn.receiveALine()
    }

    @Throws(Exception::class)
    override fun closeConnection() {
        conn.closeConnection()
    }
}